dbus-codegen/src/main.rs

```diff
@@ -47,15 +47,22 @@ Defaults to 'RefClosure'."))
              .help("If present, skips a specific prefix for interface names, e g 'org.freedesktop.DBus.'."))
         .arg(clap::Arg::with_name("futures").short("f").long("futures")
              .help("Generates code to use with futures 0.3 (experimental)"))
+        .arg(clap::Arg::with_name("file").required(false).help("D-Bus XML Introspection file"))
         .get_matches();
 
+    if matches.value_of("destination").is_some() && matches.value_of("file").is_some() {
+        panic!("Expected either xml file path as argument or destination option. But both are provided.");
+    }
+	
     let s = 
     if let Some(dest) = matches.value_of("destination") {
         let path = matches.value_of("path").unwrap_or("/");
         let bus = if matches.is_present("systembus") { dbus::BusType::System } else { dbus::BusType::Session };
         let c = dbus::Connection::get_private(bus).unwrap();
         let p = c.with_path(dest, path, 10000);
         p.introspect().unwrap()
+    } else if let Some(file_path) = matches.value_of("file")  {
+        std::fs::read_to_string(file_path.to_string()).unwrap()
     } else {
```

Comment:
I'm very sorry but i only saw this `else` block now, which takes the interface XML from stdin if neither `destination` nor `file` arguments are specified. So I'm afraid we need to remove the check above for either `destination` or `file` being specified.

Reply:
sorry, my bad.

Reply:
@zeenix In the above check we are checking whether both destination and file are provided. If both are provided we panic. that's fine isn't it? because if both are provided it is ambiguous what to use?

Reply:
Ah yes, sorry I had a long day but good that you can point out when I'm wrong. We'll need that quality. :)